ghost/core/core/server/services/stats/ReferrersStatsService.js

```
@@ -455,6 +455,136 @@ class ReferrersStatsService {
+     * @param {string} [options.post_id] - Optional filter by post ID
+     * @returns {Promise<{data: UtmGrowthStat[], meta: {}}>}
+     */
+    async getUtmGrowthStats(options = {}) {
+        const utmField = options.utm_type || 'utm_source';
+        const limit = options.limit || 50;
+        const postId = options.post_id;
+
+        // Validate utm_type is a valid UTM field
+        const validUtmFields = ['utm_source', 'utm_medium', 'utm_campaign', 'utm_term', 'utm_content'];
+        if (!validUtmFields.includes(utmField)) {
+            throw new Error(`Invalid utm_type: ${utmField}. Must be one of: ${validUtmFields.join(', ')}`);
```

ESLint

[error] Direct calls to new Error() are not allowed. Please use @tryghost/errors.

(ghost/ghost-custom/no-native-error)